python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -932,6 +932,29 @@ def _ensure_persisted_execution_plan_snapshot(
 
         return execution_plan_snapshot_id
 
+    def _asset_materialization_planned_events(self, pipeline_run, execution_plan_snapshot):
+        from dagster.core.events import DagsterEvent
+        from dagster.core.events.log import EventLogEntry
+
+        pipeline_name = pipeline_run.pipeline_name
+
+        for step in execution_plan_snapshot.steps:
+            if step.key in execution_plan_snapshot.step_keys_to_execute:
+                for output in step.outputs:
+                    asset_key = output.properties.asset_key
+                    if asset_key:
+                        event = DagsterEvent.asset_materialization_planned(pipeline_name, asset_key)
+                        event_record = EventLogEntry(
+                            user_message="",
+                            level=logging.DEBUG,
```

Comment:
might be good to leave reasoning next to the debug log level setting

Reply:
Okay, added a comment next to the setting